test/extended/networking/egress_firewall.go

```diff
@@ -0,0 +1,160 @@
+	OpenShiftSDNManifest = "sdn-egressnetworkpolicy-test.yaml"
+)
+
+var _ = g.Describe("[sig-network][Feature:EgressFirewall]", func() {
+
+	egFwoc := exutil.NewCLI(EgressFWE2E)
+	egFwf := egFwoc.KubeFramework()
+
+	// The OVNKubernetes subnet plugin supports EgressFirewall objects.
+	InOVNKubernetesContext(
+		func() {
+			g.It("Should ensure egressfirewall is created", func() {
+				doEgressFwTest(egFwf, egFwoc, OVNKManifest)
+			})
+		},
+	)
+	// For Openshift SDN its supports EgressNetworkPolicy objects
+	InOpenShiftSDNContext(
+		func() {
+			g.It("Should ensure egressnetworkpolicy is created", func() {
+				doEgressFwTest(egFwf, egFwoc, OpenShiftSDNManifest)
+			})
+		},
+	)
+	noegFwoc := exutil.NewCLI(NoEgressFWE2E)
+	noegFwf := noegFwoc.KubeFramework()
+	g.It("EgressFirewall should have no impact outside its namespace", func() {
+		g.By("Creating test pod")
+		pod := "dummy"
+		o.Expect(createTestEgressFw(noegFwf, pod)).To(o.Succeed())
+		g.By("Sending traffic should all pass with no egress firewall impact")
+		_, err := noegFwoc.Run("exec").Args(pod, "--", "ping", "-c", "1", "8.8.8.8").Output()
+		expectNoError(err)
+
+		_, err = noegFwoc.Run("exec").Args(pod, "--", "ping", "-c", "1", "1.1.1.1").Output()
+		expectNoError(err)
+
+		_, err = noegFwoc.Run("exec").Args(pod, "--", "curl", "-q", "-s", "-I", "-m1", "https://docs.openshift.com").Output()
+		expectNoError(err)
+
+	})
+})
+
+func doEgressFwTest(f *e2e.Framework, oc *exutil.CLI, manifest string) error {
+	g.By("Creating test pod")
+	o.Expect(createTestEgressFw(f, EgressFWTestPod)).To(o.Succeed())
+
+	g.By("Creating an egressfirewall object")
+	egFwYaml := exutil.FixturePath("testdata", "egress-firewall", manifest)
+
+	g.By(fmt.Sprintf("Calling oc create -f %s", egFwYaml))
+	err := oc.AsAdmin().Run("create").Args("-f", egFwYaml).Execute()
+	o.Expect(err).NotTo(o.HaveOccurred(), "created egress-firewall object")
+
+	o.Expect(sendEgressFwTraffic(f, oc, EgressFWTestPod)).To(o.Succeed())
+
+	g.By("Deleting test pod")
+	deleteTestEgressFw(f)
+	return err
+}
+
+func sendEgressFwTraffic(f *e2e.Framework, oc *exutil.CLI, pod string) error {
+	// Test ICMP / Ping to Google’s DNS IP (8.8.8.8) should pass
+	// because we have allow cidr rule for 8.8.8.8
+	g.By("Sending traffic that matches allow cidr rule")
+	_, err := oc.Run("exec").Args(pod, "--", "ping", "-c", "1", "8.8.8.8").Output()
+	expectNoError(err)
+
+	// Test ICMP / Ping to Cloudfare DNS IP (1.1.1.1) should fail
+	// because there is no allow cidr match for 1.1.1.1
+	g.By("Sending traffic that does not matche allow cidr rule")
+	_, err = oc.Run("exec").Args(pod, "--", "ping", "-c", "1", "1.1.1.1").Output()
+	expectError(err)
+
+	// Test curl to docs.openshift.com should pass
+	// because we have allow dns rule for docs.openshift.com
+	g.By("Sending traffic that matches allow dns rule")
+	_, err = oc.Run("exec").Args(pod, "--", "curl", "-q", "-s", "-I", "-m1", "https://docs.openshift.com").Output()
+	expectNoError(err)
+	return nil
```

Comment:
Maybe adding some failing curl test case can be useful here?

Reply:
the rule in the fw policy in the yaml is to allow dns rule for the above https://docs.openshift.com  are u suggesting replace that with deny instead of allow to fail the curl ?

Reply:
I mean adding a test similar to "Sending traffic that does not matche allow cidr rule" but for dns. 
So curl any denied dns and check that curl fails (with current fw policy).
Makes sense?

Reply:
sure I will add another curl to different host and it should fail